python/cuml/test/test_linear_model.py

```diff
@@ -83,6 +83,11 @@ def make_classification_dataset(datatype, nrows, ncols, n_info, num_classes):
 )
 def test_linear_regression_model(datatype, algorithm, nrows, column_info):
 
+    if algorithm == "svd" and nrows > 46340:
+        pytest.skip("svd solver is not supported for the data that has more"
+                     "than 46340 samples (rows) if you are using CUDA version"
+                     "10.1")
```

Comment:
That is true only for CUDA 10.1? Shouldn't we add a check for the cuda version then as well?

Reply:
The versioning error message, I copied from cusolver which gets flushed to stderr. While CUDA 11 does change to 64-bit addressing, the fix will come once we change our algorithms to `size_t` from `int`, right?

Reply:
totally true, so I would add that to the message so that it is accurate, mainly since these stress tests are for us and that info will be useful for someone looking at them for the first time

Reply:
Do you guys know where the 46,340 comes from? 

I'd have thought this error would have been based on both rows and columns, not just rows. It seems strange to me that they'd set a hard limit on the number of rows but not on the number of columns (or that one wouldn't constrain the other).

Reply:
Just wondering if the constraint here should actually be `n_rows * n_cols > 2^31`. Also, if this really is because of the 32-bit vs 64-bit data size limit, wouldn't we have this issue in 10.2 as well? We haven't switched over to use the new APIs yet. I checked out the cusolver codebase and grep isn't returning anything for 46340.

Reply:
Woops, looks like we ourselves are throwing this error. Previous dev must have tested some limits? Comes from 'svd.cuh' at line 66

Reply:
Nevermind- the limit is imposed by cuml from the left eigenvectors. I suppose for an extremely wide matrix the limit would need to be the number of columns. I was thinking it was the input but it's the output that's imposing the limit.

Reply:
Will this limit be increased in CUDA 11, or is it there to remain?

Reply:
CUDA 11 provides [cuSolverDnGesvd](https://docs.nvidia.com/cuda/cusolver/index.html#cuSolverDnGesvd), which takes `int64_t` for describing the dimensions. CUDA 10.2 cuSolver has the same limitation as 10.1 AFAIK.

Reply:
Updated the doc for `10.x`